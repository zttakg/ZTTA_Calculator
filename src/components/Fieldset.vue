<script>
export default {
  name: "Fieldset",

  props: {
    legend: String
  },

  // eslint-disable-next-line no-unused-vars
  render(h) {
    return (
      <fieldset class="fieldset">
        {this.legend && <legend class="fieldset__legend">{this.legend}</legend>}
        {this.$slots.default &&
          this.$slots.default.map(child => (
            <div class="fieldset__item">{child}</div>
          ))}
      </fieldset>
    );
  }
};
</script>
