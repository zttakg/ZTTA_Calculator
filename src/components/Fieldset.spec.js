import { shallowMount } from "@vue/test-utils";
import Fieldset from "./Fieldset";

describe("<Fieldset /> component", () => {
  const slots = {
    default: [
      "<input type='radio' name='test' id='r1' />",
      "<input type='radio' name='test' id='r2' />"
    ]
  };
  it("should render correct without props", () => {
    const wrapper = shallowMount(Fieldset, {
      slots
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render all props", () => {
    const wrapper = shallowMount(Fieldset, {
      propsData: {
        legend: "Test"
      },
      slots
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render with one child in default slot", () => {
    const wrapper = shallowMount(Fieldset, {
      propsData: {
        legend: "Test"
      },
      slots: {
        default: "<input type='radio' name='test' id='r1' />"
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render without child in default slot", () => {
    const wrapper = shallowMount(Fieldset, {
      propsData: {
        legend: "Test"
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
