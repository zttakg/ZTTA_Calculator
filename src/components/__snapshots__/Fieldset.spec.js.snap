// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fieldset /> component should render all props 1`] = `
<fieldset class="fieldset">
  <legend class="fieldset__legend">Test</legend>
  <div class="fieldset__item">
    <input type="radio" name="test" id="r1">
  </div>
  <div class="fieldset__item">
    <input type="radio" name="test" id="r2">
  </div>
</fieldset>
`;

exports[`<Fieldset /> component should render correct without props 1`] = `
<fieldset class="fieldset">
  <div class="fieldset__item">
    <input type="radio" name="test" id="r1">
  </div>
  <div class="fieldset__item">
    <input type="radio" name="test" id="r2">
  </div>
</fieldset>
`;

exports[`<Fieldset /> component should render with one child in default slot 1`] = `
<fieldset class="fieldset">
  <legend class="fieldset__legend">Test</legend>
  <div class="fieldset__item">
    <input type="radio" name="test" id="r1">
  </div>
</fieldset>
`;

exports[`<Fieldset /> component should render without child in default slot 1`] = `
<fieldset class="fieldset">
  <legend class="fieldset__legend">Test</legend>
</fieldset>
`;
